external/wakaama/examples/client/lwm2mclient.c

```diff
@@ -1676,148 +1016,554 @@ static void process_udpconn(int sockfd, fd_set *readfds, client_data_t data)
+	char *name = "testlwm2mclient";
+	int lifetime = 300;
+#endif
+	char *server = g_serverAddr;
+	char *serverPort = g_serverPort;
+	char *name = "myawesomeclient";
+	int lifetime = g_lifetime;
+	int batterylevelchanging = 0;
+	time_t reboot_time = 0;
+	int opt;
+	bool bootstrapRequested = g_bootstrapRequested;
+	bool serverPortChanged = false;
+
+	if (bootstrapRequested) {
+		server = g_bootstrapserverAddr;
+		serverPort = g_bootstrapserverPort;
 	} else {
-		fprintf(stderr, "Error calling object dump\n");
+		server = g_serverAddr;
+		serverPort = g_serverPort;
 	}
-	return -1;
-}
-
-/* Private APIs */
 
-static int read_input_command_line(char *buf)
-{
-	char buffer[MAX_PACKET_SIZE] = {0,};
 
-	int nbytes = 0;
-	int pos = 0;
-	int char_idx = 0;
-	int bufsize = MAX_PACKET_SIZE;
+#ifdef LWM2M_BOOTSTRAP
+	lwm2m_client_state_t previousState = STATE_INITIAL;
+#endif
 
-	do {
-		nbytes = read(STDIN_FILENO, &buffer[pos], (bufsize - pos));
-		if (nbytes <= 0) {
-			fprintf(stderr, "cannot read command\n");
+	char *pskId = NULL;
+	char *psk = NULL;
+	uint16_t pskLen = -1;
+	char *pskBuffer = NULL;
+
+	/*
+	 * The function start by setting up the command line interface (which may or not be useful depending on your project)
+	 *
+	 * This is an array of commands describes as { name, description, long description, callback, userdata }.
+	 * The firsts tree are easy to understand, the callback is the function that will be called when this command is typed
+	 * and in the last one will be stored the lwm2m context (allowing access to the server settings and the objects).
+	 */
+	command_desc_t commands[] = {
+		{"list", "List known servers.", NULL, prv_output_servers, NULL},
+		{
+			"change", "Change the value of resource.", " change URI [DATA]\r\n"
+			"   URI: uri of the resource such as /3/0, /3/0/2\r\n"
+			"   DATA: (optional) new value\r\n", prv_change, NULL
+		},
+		{
+			"update", "Trigger a registration update", " update SERVER\r\n"
+			"   SERVER: short server id such as 123\r\n", prv_update, NULL
+		},
+#ifdef LWM2M_BOOTSTRAP
+		{"bootstrap", "Initiate a DI bootstrap process", NULL, prv_initiate_bootstrap, NULL},
+		{
+			"dispb", "Display current backup of objects/instances/resources\r\n"
+			"\t(only security and server objects are backupped)", NULL, prv_display_backup, NULL
+		},
+#endif
+		{"ls", "List Objects and Instances", NULL, prv_object_list, NULL},
+		{"disp", "Display current objects/instances/resources", NULL, prv_display_objects, NULL},
+		{
+			"dump", "Dump an Object", "dump URI"
+			"URI: uri of the Object or Instance such as /3/0, /1\r\n", prv_object_dump, NULL
+		},
+		{"add", "Add support of object 1024", NULL, prv_add, NULL},
+		{"rm", "Remove support of object 1024", NULL, prv_remove, NULL},
+		{"quit", "Quit the client gracefully.", NULL, prv_quit, NULL},
+		{"^C", "Quit the client abruptly (without sending a de-register message).", NULL, NULL, NULL},
+
+		COMMAND_END_LIST
+	};
+
+	printf("Inside client_main check serial\n");
```

Comment:
I'm not sure but this code will be used in Device management.
So we added a dependency such as "#if !defined(CONFIG_DM)".

You can figure out it in latest code. Thanks.